packages/plugin-ext/src/main/browser/custom-editors/plugin-custom-editor-registry.ts

```diff
@@ -0,0 +1,142 @@
+/********************************************************************************
+ * Copyright (c) 2021 SAP SE or an SAP affiliate company and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject, postConstruct } from 'inversify';
+import { CustomEditor } from '../../../common';
+import { Disposable, DisposableCollection } from '@theia/core/lib/common/disposable';
+import { CustomEditorOpener } from './custom-editor-opener';
+import { WorkspaceCommands } from '@theia/workspace/lib/browser';
+import { CommandRegistry, Emitter, MenuModelRegistry } from '@theia/core';
+import { SelectionService } from '@theia/core/lib/common';
+import { UriAwareCommandHandler } from '@theia/core/lib/common/uri-command-handler';
+import { NavigatorContextMenu } from '@theia/navigator/lib//browser/navigator-contribution';
+import { ApplicationShell, DefaultOpenerService, WidgetManager } from '@theia/core/lib/browser';
+import { CustomEditorWidget } from './custom-editor-widget';
+
+@injectable()
+export class PluginCustomEditorRegistry {
+    private readonly editors = new Map<string, CustomEditor>();
+    private readonly pendingEditors = new Set<CustomEditorWidget>();
+    private readonly resolvers = new Map<string, (widget: CustomEditorWidget) => void>();
+
+    private readonly onPendingOpenCustomEditorEmitter = new Emitter<string>();
+    readonly onPendingOpenCustomEditor = this.onPendingOpenCustomEditorEmitter.event;
```

Comment:
sorry, I didn't catch from the event name what the event means
could you clarify what will happen when the event is fired

Reply:
Change to `onWillOpenCustomEditor`. let me know if clearer. Thanks